lib/spack/spack/reporters/cdash.py

```diff
@@ -123,11 +124,15 @@ def __init__(self, configuration: CDashConfiguration):
         self.multiple_packages = False
 
     def report_build_name(self, pkg_name):
-        return (
+        buildname = (
             "{0} - {1}".format(self.base_buildname, pkg_name)
             if self.multiple_packages
             else self.base_buildname
         )
+        if len(buildname) > 190:
+            warnings.warn("Build name exceeds CDash 191 character maximum and will be truncated.")
```

Comment:
The limit is technically 190.

```suggestion
            warnings.warn("Build name exceeds CDash 190 character maximum and will be truncated.")
```